subprojects/launcher/src/main/java/org/gradle/launcher/exec/BuildOutcomeReportingBuildActionRunner.java

```diff
@@ -57,8 +62,35 @@ public Result run(BuildAction action, BuildController buildController) {
 
         Result result = delegate.run(action, buildController);
 
+        Throwable failure = SingleMessageLogger.getDeprecationFailure();
+        if (failure != null) {
+            // Replace result if we have fail on warning
+            result = computeUpdatedResult(result, failure);
+        }
+
         buildLogger.logResult(result.getBuildFailure());
         new TaskExecutionStatisticsReporter(styledTextOutputFactory).buildFinished(taskStatisticsCollector.getStatistics());
         return result;
     }
+
+    private Result computeUpdatedResult(Result previousResult, Throwable deprecationFailure) {
+        if (previousResult.getBuildFailure() != null) {
+            // Enhance already reported failures
+            Throwable buildFailure = previousResult.getBuildFailure();
+            ArrayList<Throwable> newFailures;
```

Comment:
can probably be weakened
```suggestion
            List<Throwable> newFailures;
```